### PR TITLE
Unit tests for ErrorController. Issue#257

### PR DESCRIPTION
**Pull request by Soupaa**
# Description
Unit tests have been written for 4/5 Error handling functions in ErrorController. The last unfinished unit test, which is for Error 404 has not been written, as the correct solution to solve this specific area cannot be identified within the given time constraint.

Therefore this pull request contains these unit tests:
1. Unit test for Error 401 Unauthorized
2. Unit test for Error 403 Forbidden
3. Unit test for Error 400 BadRequest
4. Unit test for Error UnknownError

Fixes #\ Part of Issue #257

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

The four unit tests has been written with xUnit. Therefore this issue is mainly about testing the existing function within ErrorController file.

The testing was conducted with the xUnit "Run Test" and "Run All Tests" buttons.

**Development Configuration**

* Flutter version: 2.0.5
* Dart version: 2.12.3

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas, if necessary
- [ ] I have made corresponding changes to the documentation, if necessary
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works, if necessary
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have Acceptance Tested this on an iOS device
- [ ] I have Acceptance Tested this on an Android device

### CODE REVIEW

File: GirafRest/appsettings.Development.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿{
     "ConnectionStrings": {
-        "DefaultConnection": "server=192.38.56.153;port=3306;userid=root;password=Giraf123;database=giraf;Allow User Variables=True"
+        "DefaultConnection": "server=localhost; port=3306; userid=user; password=password; database=giraf; Allow User Variables=True"
```
Comment (NikolajKrogh) [inline GirafRest/appsettings.Development.json]: Det er nok en fejl at i har sat server connection til localhost og ændret password

File: GirafRest.Test/Controllers/ErrorControllerTest.cs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+{
+    public class ErrorControllerTest
+    {
+        [Fact]
+       public void Unauthorized401_Test()
+       {
+            // Arrange
+            int status = StatusCodes.Status401Unauthorized;
+            var mockFeatureCollection = new Mock<IFeatureCollection>();
+            mockFeatureCollection.Setup(f => f.Get<IStatusCodeReExecuteFeature>())
```
Comment (NikolajKrogh) [inline GirafRest.Test/Controllers/ErrorControllerTest.cs]: f står for IFeatureCollection ikke? Hvis det gør kan det så ikke få et mere sigende navn så som feature?

Comment (NikolajKrogh) [reply GirafRest.Test/Controllers/ErrorControllerTest.cs]: Kan se det også sker andre steder

**Review by NikolajKrogh — CHANGES_REQUESTED** [review]

File: GirafRest.Test/Controllers/ErrorControllerTest.cs (hunk 2 above)
Comment (Soupaa) [reply GirafRest.Test/Controllers/ErrorControllerTest.cs]: Jo, det kigger jeg på hurtigest muligt

**Comment by makelblack159** [comment]
kan vi få den fixed så den ikke failer integration test ?

File: GirafRest.Test/Controllers/ErrorControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+            // Assert
+            Assert.Equal(statusCodeResult, status);
+       }
+
+        [Fact]
+       public void BadRequest400_Test()
+       {
+            // Arrange
+            int status = 0;
```
Comment (makelblack159) [inline GirafRest.Test/Controllers/ErrorControllerTest.cs]: giver den ikke en status 400 og ikke 0 ?

**Comment by JonasStjerne** [comment]
> kan vi få den fixed så den ikke failer integration test ?

Unit test burde ikke have noget med integration test at gøre. Der bliver kigget på integration test en del det her sprint ved jeg. Der sker en fejl i dem ved connection til database.

**Review by makelblack159 — APPROVED** [review]

File: GirafRest.Test/Controllers/ErrorControllerTest.cs (hunk 3 above)
Comment (Soupaa) [reply GirafRest.Test/Controllers/ErrorControllerTest.cs]: Jo, det har du ret i. Den er fikset

**Review by Soupaa — COMMENTED** [review]
Changed the json back to dev values

**Review by NikolajKrogh — APPROVED** [review]

**Pull request merged**